### PR TITLE
syntax, interp: support brace expansion in declaration commands

### LINKED ISSUE #1257: syntax: brace expansion not supported in declaration variable names
Brace expansion is not applied in declaration commands (`local`, `export`, `declare`), causing a parse error.

### Reproduction

```bash
$ bash -c 'f() { local {A,B}_VAR=1; echo "A=$A_VAR B=$B_VAR"; }; f'
A=1 B=1

$ gosh -c 'f() { local {A,B}_VAR=1; echo "A=$A_VAR B=$B_VAR"; }; f'
1:13: invalid var name
```

Bash expands braces before parsing the assignment, producing `local A_VAR=1 B_VAR=1`. Since declaration commands are parsed as keywords, the parser validates variable names at parse time — before brace expansion can occur at runtime.

Note that brace expansion in argument position works correctly (`echo {A,B}_VAR` outputs `A_VAR B_VAR`). The issue is specific to variable names in declarations.

### Real-world impact

Gentoo ebuilds use this pattern:

```bash
# sys-apps/findutils-4.10.0.ebuild:84
local -x RUN_{VERY_,}EXPENSIVE_TESTS=$(usex test-full yes no)

# sys-apps/sed-4.9-r1.ebuild:61
local -x RUN_{VERY_,}EXPENSIVE_TESTS=$(usex test-full yes no)
```

Because this is a parse error, the entire script fails — not just the function containing the pattern.

### Context

This limitation is noted in the README caveats. This was item 5 from #1252, filed separately as requested.

### PR DESCRIPTION
**Pull request by kolkov**
I took the liberty of submitting this PR because the fix turned out to be minimal (one-line parser change), but I fully understand if you'd prefer a different approach or consider this a deliberate design trade-off.

The parser currently errors on `local {A,B}_VAR=1` because `declClause` validates variable names before brace expansion can occur at runtime. The fix skips the "invalid var name" error when the name part contains `{`, letting it fall through to `getWord()`. At runtime, `flattenAssigns` already handles the rest: `Fields` calls `SplitBraces`/`Braces` to expand `{A,B}_VAR=1` into `A_VAR=1 B_VAR=1`, and the existing `ValidName` check catches invalid names after expansion.

This pattern appears in Gentoo ebuilds:

```bash
local -x RUN_{VERY_,}EXPENSIVE_TESTS=$(usex test-full yes no)
```

Changes:
- `syntax/parser.go`: skip error when name before `=` contains `{`
- `syntax/filetests_test.go`: parser AST test
- `interp/interp_test.go`: 4 runtime tests
- `README.md`: remove caveat

Fixes #1257.

### CODE REVIEW

**Review by mvdan — COMMENTED** [review]
Thanks for the suggestion, this is neat. I was thinking about regular assignments, and not `declare`, so I was forgetting that making braces work in `declare` is actually pretty easy.

Some suggestions below to make the edge cases align with Bash too.

File: interp/interp_test.go [hunk 1]
```diff
@@ -3289,6 +3289,12 @@ done <<< 2`,
 	{"b=c; echo a{1,2}$b", "a1c a2c\n"},
 	{"echo a{1,2}'bc'", "a1bc a2bc\n"},
 
+	// brace expansion in declarations
+	{"f() { local {A,B}_VAR=1; echo $A_VAR $B_VAR; }; f", "1 1\n"},
+	{"f() { declare {x,y}=val; echo $x $y; }; f", "val val\n"},
+	{"f() { local -x RUN_{VERY_,}EXPENSIVE_TESTS=yes; echo $RUN_EXPENSIVE_TESTS; }; f", "yes\n"},
+	{"f() { local {A,B}_VAR; A_VAR=1; B_VAR=2; echo $A_VAR $B_VAR; }; f", "1 2\n"},
```
[the remark this answers is not in the document]
Comment (kolkov) [reply interp/interp_test.go]: Done. Changed all tests to use `declare` without the function wrapper, and added the edge cases:

- `declare {foo=x,bar=y}` — works, `flattenAssigns` splits into `foo=x` and `bar=y`
- `declare foo{bar=baz` — runtime error, no closing `}` so no brace expansion, `ValidName("foo{bar")` fails
- `declare {x,y}=(1 2)` — parse error, added in `parser_test.go` since the parser rejects `(` after the naked word
- `{a,b}=value` — treated as regular words (command not found), same as Bash

All verified against Bash. Force-pushed.

**Review by mvdan — APPROVED** [review]
